storage/range.go

```diff
@@ -614,7 +613,13 @@ func (r *Range) addReadOnlyCmd(ctx context.Context, args proto.Request, reply pr
 		if header.Timestamp.Equal(proto.ZeroTimestamp) {
 			header.Timestamp = r.rm.Clock().Now()
 		}
-		intents, err := r.executeCmd(r.rm.Engine(), nil, args, reply)
+		executedReply, intents, err := r.executeCmd(r.rm.Engine(), nil, args)
+		if executedReply.Header().Error != nil {
+			panic("the world is on fire")
+		}
+		executedReply.Header().SetGoError(err)
+		reply.Reset()
+		gogoproto.Merge(reply, executedReply)
```

Comment:
some overhead here and the escape analysis for `executedReply` likely won't look good. What do you win by doing this?